### PR TITLE
Remove codicon.css, document and explain future usage

### PR DESCRIPTION
**Pull request by kaisalmen**
- This removes the need to have codicon as depedency to the sprotty core
- Replace rimraf wih shx along the way

This is the proposed solution for #311

### CODE REVIEW

File: packages/sprotty/build/src/buildCodicon.ts [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+
+        fs.readFile('../../node_modules/@vscode/codicons/dist/codicon.ttf', 'base64', (errTtf, dataTtf) => {
+            if (errTtf) {
+                console.error(errTtf);
+                return;
+            }
+            const ttfData = `data:font/ttf;base64,${dataTtf}`;
+            const alteredCssData = cssData.replace('"./codicon.ttf?0e5b0adf625a37fbcd638d31f0fe72aa"', ttfData);
+
+            fs.writeFileSync('./css/generated/codicon.css', alteredCssData);
```
Comment (spoenemann) [inline packages/sprotty/build/src/buildCodicon.ts:25]: This is a redistribution of the codicon content and so we need to attribute that correctly according to their license:

https://github.com/microsoft/vscode-codicons/blob/e7d905a9cb4f06a0d4c4c93279967734faf66f0f/LICENSE#L217-L244

**Review by spoenemann — COMMENTED** [review]
I'm ok with the changes (but see comments below). @ndoschek @planger @tortmayr please have a look as well.

This brings up a more general question to me regarding the original introduction of the codicons dependency in #249: Shouldn't this be a dependency of the concrete application or downstream library rather than the core package? The sprotty package itself does not need this dependency, only the applications that want to include such icons do. A lot of users of the sprotty package do not need codicons. Would it be feasible to drop the dependency and leave it to the responsibility of the package users to bundle the icons if they need them?

File: packages/sprotty/build/tsconfig.json [hunk 2]
```diff
@@ -0,0 +1,16 @@
+{
```
[the remark this answers is not in the document]
Comment (kaisalmen) [reply packages/sprotty/build/tsconfig.json:1]: Yes, that is a good idea, plus the "debug" functionality (`generateAlernateLoadingFunction` and storage in separate files) could be removed, too.

File: packages/sprotty/build/src/buildCodicon.ts (hunk 1 above)
Comment (kaisalmen) [reply packages/sprotty/build/src/buildCodicon.ts:25]: Yes, thank you, I didn't check that, yet.

**Comment by tortmayr** [comment]
> This brings up a more general question to me regarding the original introduction of the codicons dependency in #249: Shouldn't this be a dependency of the concrete application or downstream library rather than the core package? The sprotty package itself does not need this dependency, only the applications that want to include such icons do. A lot of users of the sprotty package do not need codicons. Would it be feasible to drop the dependency and leave it to the responsibility of the package users to bundle the icons if they need them?

In general, I agree this probably should be a dependency of the downstream project. This way the consuming project/library could is also in complete control of the codicon version that should be used.

>The sprotty package itself does not need this dependency, only the applications that want to include such icons do.

That's currently not completely true. It seems like default implementation of  the command palette is currently using codicons.
So we should probably refactor this or at least mention in the ts doc that codicon is needed for the command palette feature.

File: packages/sprotty/build/tsconfig.json (hunk 2 above)
[the remark this answers is not in the document]
Comment (kaisalmen) [reply packages/sprotty/build/tsconfig.json]: Transformed it to JavaScript and removed everything unnecessary.

**Comment by spoenemann** [comment]
@tortmayr @planger so shall we work towards getting rid of the codicons dependency in the core package? Of course this needs to be well documented in the CHANGELOG, but it sounds viable to me.

**Comment by spoenemann** [comment]
@tortmayr ping

**Comment by kaisalmen** [comment]
I have squashed and force pushed the branch on current main and performed a bit of clean-up.

We could use this workaround for now until we find complete solution. It would help prevent for example this problem: https://github.com/eclipse/sprotty/issues/311#issuecomment-1323416308

**Review by spoenemann — CHANGES_REQUESTED** [review]
I think we can go right to the cleaner solution instead of introducing a workaround.

My proposal:
 * Remove the dependency in package.json
 * Remove this import: https://github.com/eclipse/sprotty/blob/d9b542a9974d7db03742010917ca64dc9d0eed76/packages/sprotty/src/features/command-palette/di.config.ts#L16
 * Document in CHANGELOG.md that those who are using the command palette need to import the codicons CSS in their app.

**Comment by tortmayr** [comment]
@spoenemann Sorry for the late reply. The proposed changes look good to me 👍🏼

**Comment by kaisalmen** [comment]
@spoenemann The rework is done. This is now ready for review. I fought with the webpack config, beacuse codicon.ttf was not loaded. In the end, removing the extra config removed the problem. webpack 5 (introduced in between initial addition of command palette) changed asset handling.

**Review by spoenemann — APPROVED** [review]
Thank You!

**Pull request merged**